src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs

```diff
@@ -13,17 +13,17 @@ namespace AutoRest.CSharp.Output.Models.Types
     internal class DpgOutputLibrary : OutputLibrary
     {
         private readonly IReadOnlyDictionary<InputEnumType, EnumType> _enums;
-        private readonly IReadOnlyDictionary<InputModelType, ModelTypeProvider> _models;
+        private readonly IReadOnlyDictionary<InputModelType, ObjectType> _models;
         private readonly bool _isCadlInput;
 
         public TypeFactory TypeFactory { get; }
         public IEnumerable<EnumType> Enums => _enums.Values;
-        public IEnumerable<ModelTypeProvider> Models => _models.Values;
+        public IEnumerable<ObjectType> Models => _models.Values;
```

Comment:
why is this change?

Reply:
Here is the inheritance relationship among them: 
```
SystemObjectType : ObjectType
ModelTypeProvider : SerializableObjectType
SerializableObjectType : ObjectType
```
And we are not actually using the elements in this array as `ModelTypeProvider` anywhere (and we should not)

Reply:
This change is not required by this PR right now. But I believe we should do this level's abstraction, therefore I still preserved this change on this PR

Reply:
What actually Type will the Models list item be?  ModelTypeProvider only or it can be ObjectType. I am afraid it will affect the Model Generation. In Model Generation, it will check the type of the model, and then choose the method to generate.

Reply:
> 

That is how polymorphism in OO works. Everything is calculated when it is constructed no matter what type of variable the object is storing into.

Reply:
This change will never introduce any break - because this is what polymorphism is for.
But I change my mind and already reverted all these changes. I could always introduce it back when it is necessary